### PR TITLE
Add genAI integrations

### PR DESCRIPTION
**Pull request by renetapopova**

### CODE REVIEW

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-cypher-882.surge.sh

New pages:
- [GenAI integrations](https://neo4j-docs-cypher-882.surge.sh/cypher-manual/5/genai-integrations)

File: modules/ROOT/pages/genai-integrations.adoc [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+    WITH n
+    WITH collect(n) AS nodes, collect(n.text) AS resources
+    CALL genai.vector.encodeBatch(resources, "VertexAI", { token: $token, projectId: $project }) YIELD index, vector
+    CALL db.create.setNodeVectorProperty(nodes[index], "vector", vector)
+} IN TRANSACTIONS OF 1000 ROWS
+----
+====
+
+[[ai-providers]]
+== AI Providers
```
Comment (nilsceberg) [inline modules/ROOT/pages/genai-integrations.adoc]: I think I personally would prefer it as "GenAI" / "Generative AI" and not just "AI" - the latter can be somewhat more controversial.

File: modules/ROOT/pages/genai-integrations.adoc [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+
+The following AI providers are supported for generating vector embeddings.
+Each provider has its own configuration map that can be passed to the `genai.vector.encode()` or `genai.vector.encodeBatch()` functions.
+
+=== Vertex AI
+
+* Identifier: `"Vertex AI"`
```
Comment (nilsceberg) [inline modules/ROOT/pages/genai-integrations.adoc]:
```suggestion
* Identifier: `"VertexAI"`
```

The heading can differ but the identifier mustn't have the space.

File: modules/ROOT/pages/genai-integrations.adoc [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+CREATE (:Page { index: index, text: resource, vector: vector })-[:OF]->(book)
+----
+====
+
+.Generate embeddings for many text properties
+====
+
+If you want to generate embeddings for the text content of all nodes with label `Tweet`, you can use `CALL ... IN TRANSACTIONS` to split the work up into batches and issue one API request per batch.
+
+Assuming nodes with the `Tweet` have a `text` property, you can generate vector embeddings for each one and write them to the `embedding` property on each one in batches of 1000, for example:
```
Comment (nilsceberg) [inline modules/ROOT/pages/genai-integrations.adoc]:
```suggestion
Assuming nodes with the `Tweet` label have a `text` property, you can generate vector embeddings for each one and write them to the `embedding` property on each one in batches of 1000, for example:
```

My typo :)

**Review by nilsceberg — APPROVED** [review]
Looks great, just a couple of tiny things!

File: modules/ROOT/pages/genai-integrations.adoc (hunk 2 above)
Comment (renetapopova) [reply modules/ROOT/pages/genai-integrations.adoc]: ah, of course

File: modules/ROOT/pages/genai-integrations.adoc [hunk 4]
```diff
@@ -0,0 +1,291 @@
...
+    CALL genai.vector.encodeBatch(resources, "VertexAI", { token: $token, projectId: $project }) YIELD index, vector
+    CALL db.create.setNodeVectorProperty(nodes[index], "vector", vector)
+} IN TRANSACTIONS OF 1000 ROWS
+----
+====
+
+[[ai-providers]]
+== GenAI providers
+
+The following AI providers are supported for generating vector embeddings.
```
Comment (nilsceberg) [inline modules/ROOT/pages/genai-integrations.adoc]: Maybe on this line as well? I'll leave it up to you :)

**Review by nilsceberg — APPROVED** [review]

**Comment by neo-technology-commit-status-publisher** [comment]
Thanks for the documentation updates.

The preview documentation has now been torn down - reopening this PR will republish it.

**Pull request merged**